segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java

```diff
@@ -865,6 +880,7 @@ private SystemSnapshotRecord readSnapshotRecord(SnapshotInfo snapshotInfo, byte[
                                                         containerId, epochToRecover.get(), fileIndexToRecover.get());
                                                 return CompletableFuture.completedFuture(null);
                                             } else {
+                                                journalsProcessed.add(systemLogName);
```

Comment:
Any reason why we don't add it directly to `pendingGarbageChunks`?

Reply:
I think the intention is to add them to garbage list only after successful replay. (In reality not able to replay journals is a critical failure and will fail the container, so most likely there won't be any GC running to delete any chunks prematuarely. But thought that this way its just guaranteed to be no GC if there is no successful replay)